lib/service/cache/redis.js

```diff
@@ -51,11 +51,22 @@ class Redis extends Service {
    * @returns {Promise}
    */
   _initSequence() {
+    // @deprecated @todo Remove those lines for Kuzzle v3 then 
+    // remove 'database' from .kuzzlerc.sample and default.config.js
+    if (this._config.database) {
+      this._config.options = { ...this._config.options, db: this._config.database };
+    }
+
     if (this._config.nodes) {
-      this.client = this._buildClusterClient();
+      this._config.clusterOptions = this._config.overrideDnsLookup
+        ? { ...this._config.clusterOptions, dnsLookup: (address, callback) => callback(null, address) }
+        : this._config.clusterOptions;
```

Comment:
I think this condition should be more verbose and less long :exploding_head: ?
```suggestion
      if (this._config.overrideDnsLookup) {
        this._config.clusterOptions.dnsLookup = (address, callback) => callback(null, address);
      }
```

Reply:
Nice catch but I'll need to keep object destructuring because `this._config.clusterOptions` can be undefined

Reply:
I figured, it is not in our style to construct properties on the fly like this. Can't we initialize this option elsewhere?
Why would the **init_sequence** modify directly the received config?

Reply:
Ok found the problem. The undefined error on `clusterOptions` can only append on in unit tests (since object is initialised in the default config). I just need to update my tests to do so.